modules/apps/data-engine/data-engine-js-components-web/src/main/resources/META-INF/resources/js/custom/form/FormView.es.js

```diff
@@ -69,14 +69,28 @@ const useFormSubmit = ({apiRef, containerRef}) => {
 							return;
 						}
 
-						Liferay.Util.submitForm(event.target);
+						const avoidSubmit =
+							event.target?.dataset.ddmFormAvoidSubmit;
+
+						if (!avoidSubmit) {
+							Liferay.Util.submitForm(event.target);
+						}
+
+						Liferay.fire('ddmFormValid', {
```

Comment:
I was wondering: why you can't just use the `ddmFormSubmit` event?

Reply:
Because ddmFormSubmit is fired with a `{formId : undefined}` data.

![image](https://user-images.githubusercontent.com/67901/117950798-da1cc080-b313-11eb-91fc-070a56b117d6.png)

Also, I assume that this `formId` is an internal data engine id and I want to receive the dom form wrapper id, not the data engine id. I want to test that the valid form [is the same that the user submits](https://github.com/liferay-data-engine/liferay-portal/pull/416/files#diff-03c72f564e9e8550c75d3162552f69947f0f326e0b759f89fee31815d177b92fR264) in DM.

Reply:
Could you please rename `formId` in order to avoid confusion with the already existing one?

Thanks in advance.

Reply:
Consider it done.

![](https://media0.giphy.com/media/C7Y7Ihbp6jlTZWOpjx/giphy.gif)